readthedocs/notifications/storages.py

```diff
@@ -0,0 +1,45 @@
+from django.contrib.messages.storage.base import Message
+from django.utils.safestring import mark_safe
+from messages_extends.storages import FallbackStorage
+from messages_extends.models import Message as PersistentMessage
+from messages_extends.constants import PERSISTENT_MESSAGE_LEVELS
+
+
+class FallbackUniqueStorage(FallbackStorage):
+
+    """Persistent message fallback storage, but only stores unique notifications
+
+    This loops through all backends to find messages to store, but will skip
+    this step if the message already exists for the user in the database.
+
+    This class also assumes that notifications set as persistent messages are
+    more likely to have HTML that should be marked safe. If the level matches a
+    persistent message level, mark the message text as safe so that we can
+    render the text in templates.
+    """
```

Comment:
Yup, I can expand more on this. Duplicated messages aren't normally an issue, showing two persistent messages would cause an issue like:

* User is shown first "Subscribe now"
* User is shown second "Subscribe now"
* User click on first "Subscribe now", dismissing the message
* User signs up for a plan
* User still has a "Subscribe now" message

We could make this logic explicit, removing duplicates on subscription or something, but still seems like good UI pattern.

I'll add this info to the docs.